spec/classes/letsencrypt_install_spec.rb

```diff
@@ -1,92 +1,101 @@
 require 'spec_helper'
 
 describe 'letsencrypt::install' do
-  let(:params) { default_params.merge(additional_params) }
-  let(:default_params) do
-    {
-      configure_epel: false,
-      package_ensure: 'installed',
-      manage_install: true,
-      manage_dependencies: true,
-      path: '/opt/letsencrypt',
-      repo: 'https://github.com/certbot/certbot.git',
-      version: 'v0.9.3',
-      package_name: 'letsencrypt'
-    }
-  end
-  let(:additional_params) { {} }
+  on_supported_os.each do |os, facts|
+    let :facts do
+      facts
+    end
+    let(:params) { default_params.merge(additional_params) }
```

Comment:
A pattern I've started to use instead is:
```ruby
let(:params) do
  {
    # ...
  }
end
```

Then when you need to override it below:
```ruby
let(:params) { super().merge(install_method: 'package') }
```